### PR TITLE
RUN-3067 downloadRuntime fix

### PR DESCRIPTION
**Pull request by rdepena**
[PR To V6 Staging](https://github.com/HadoukenIO/core/pull/79)

Test results: pending.

@wenjunche @StevenEBarbaro @HarsimranSingh @whyn07m3 @datamadic @joneit please review.

### CODE REVIEW

**Comment by rdepena** [comment]
[Test results](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59527af80c101702173658b4)

File: src/browser/core_state.ts [hunk 1]
```diff
@@ -211,8 +211,8 @@ export function getUuidBySourceUrl(sourceUrl: string): string|boolean {
 
 export function getConfigUrlByUuid(uuid: string): string|boolean {
     let app = appByUuid(uuid);
-    while (app && app.appObj && app.appObj.parentUuid) {
-        app = appByUuid(app.appObj.parentUuid);
+    while (app && app.appObj && app.parentUuid) {
```
Comment (joneit) [inline src/browser/core_state.ts]: ` && app.appObj` unneeded (?)

Comment (rdepena) [inline src/browser/core_state.ts]: for sure not, good catch.

File: src/browser/core_state.ts [hunk 2]
```diff
@@ -211,7 +212,7 @@ export function getUuidBySourceUrl(sourceUrl: string): string|boolean {
 
 export function getConfigUrlByUuid(uuid: string): string|boolean {
     let app = appByUuid(uuid);
-    while (app && app.appObj && app.parentUuid) {
+    while (app && app.parentUuid) {
         app = appByUuid(app.parentUuid);
```
Comment (whyn07m3) [inline src/browser/core_state.ts]: Suggest using `getAppAncestor` method here

Comment (rdepena) [inline src/browser/core_state.ts]: sounds good. will make that change.

**Comment by rdepena** [comment]
@whyn07m3 @joneit changes made.

**Comment by HarsimranSingh** [comment]
@rdepena Can you rebase and kick off another test run?

**Comment by whyn07m3** [comment]
Can a test be created for this fix, to ensure no future regressions?

File: src/browser/core_state.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (HarsimranSingh) [inline src/browser/core_state.ts]: Is this blank line going to mess up the licensing task?

**Pull request merged**